src/TuDa.CIMS.Api/Repositories/PurchaseRepository.cs

```diff
@@ -0,0 +1,162 @@
+﻿using Microsoft.EntityFrameworkCore;
+using TuDa.CIMS.Api.Database;
+using TuDa.CIMS.Api.Interfaces;
+using TuDa.CIMS.Shared.Dtos;
+using TuDa.CIMS.Shared.Entities;
+
+namespace TuDa.CIMS.Api.Repositories;
+
+public class PurchaseRepository : IPurchaseRepository
+{
+    private readonly CIMSDbContext _context;
+
+    public PurchaseRepository(CIMSDbContext context)
+    {
+        _context = context;
+    }
+
+    /// <summary>
+    /// Returns all purchases from the database.
+    /// </summary>
+    /// <returns></returns>
+    public async Task<IEnumerable<Purchase>> GetAllAsync()
+    {
+        return await _context.Purchases
+            .Include(i => i.WorkingGroup)
+            .Include(x => x.Entries)
+            .ToListAsync();
+    }
+
+    /// <summary>
+    /// Returns a single purchase by its specific ID.
+    /// </summary>
+    /// <param name="id">teh specific ID for the purchase</param>
+    /// <returns></returns>
+    public async Task<Purchase?> GetOneAsync(Guid id)
+    {
+        return await _context
+            .Purchases
+            .Where(i => i.Id == id)
+            .Include(i => i.WorkingGroup)
+            .Include(x => x.Entries)
+            .SingleOrDefaultAsync();
+    }
+
+    /// <summary>
+    /// Updates an existing purchase by its ID using the provided update model.
+    /// </summary>
+    /// <param name="id">the specific ID of the purchase</param>
+    /// <param name="updateModel">the model containing the update values for the purchase </param>
+    /// <returns></returns>
+    public async Task<ErrorOr<Updated>> UpdateAsync(Guid id, UpdatePurchaseDto updateModel)
+    {
+        var existingItem = await _context
+            .Purchases
+            .Include(purchase => purchase.WorkingGroup)
+            .Include(y => y.Entries)
+            .Include(y => y.Buyer)
+            .SingleOrDefaultAsync(i => i.Id == id);
+
+        if (existingItem is null)
+        {
+            return Error.NotFound("Purchase.update", $"Purchase with ID {id} was not found.");
+        }
+
+        existingItem.WorkingGroup = updateModel.WorkingGroup ?? existingItem.WorkingGroup;
+        existingItem.Buyer = updateModel.Buyer ?? existingItem.Buyer;
+        existingItem.Signature = updateModel.Signature ?? existingItem.Signature;
+        existingItem.Entries = updateModel.Entries ?? existingItem.Entries;
+
+        if (updateModel.WorkingGroup is not null)
+        {
+            var workingGroup =
+                await _context.WorkingGroups.SingleOrDefaultAsync(r => r.Id == updateModel.WorkingGroup.Id);
+            if (workingGroup is null)
+            {
+                return Error.NotFound("Purchase.update",
+                    $"WorkingGroup with ID {updateModel.WorkingGroup.Id} was not found.");
+            }
```

Comment:
⛏️ Could be done by `Any`
```suggestion               
            if (!await _context.WorkingGroups.AnyAsync(r => r.Id == updateModel.WorkingGroup.Id))
            {
                return Error.NotFound("Purchase.update",
                    $"WorkingGroup with ID {updateModel.WorkingGroup.Id} was not found.");
            }
```

Reply:
I guess I can't do it like this because I have to assign 'existingItem.WorkingGroup = workingGroup;'. Correct me if I'm wrong.

Reply:
Yeah, you're right. That's my fault